src/battle/action_cmd/mega_shock.c

```diff
@@ -1,13 +1,372 @@
+extern HudElementAnim D_80108B80;
+extern HudElementAnim D_80292374;
 
-INCLUDE_ASM(s32, "battle/action_cmd/mega_shock", func_802A92A0_42DCB0);
+void btl_set_popup_duration(s32);
+void func_80268C9C(void);
+void func_80269160(void);
+void sfx_adjust_env_sound_params(s32, s32, s32, s32);
 
-INCLUDE_ASM(s32, "battle/action_cmd/mega_shock", func_802A97F8_42E208);
+void func_80268770(s32, s32, s16);
+void func_8026880C(s32, s32, s16);
 
-INCLUDE_ASM(s32, "battle/action_cmd/mega_shock", func_802A98E0_42E2F0);
+typedef struct structD_802A9930_42E340 {
+    /* 0x00 */ s16 unk_0;
+    /* 0x08 */ u16 unk_8;
+} structD_802A9930_42E340;
+
+structD_802A9930_42E340 D_802A9930_42E340[] = {
+    { 0, 0, },
+    { 0, 0x19 },
+    { 0, 0x32 },
+    { 0, 0x4B },
+    { 0, 0x4B },
+    { 0, 0 },
+};
+
+ApiStatus func_802A9000_42DA10(Evt* script, s32 isInitialCall) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    ActionCommandStatus* actionCommandStatus = &gActionCommandStatus;
+    s32 elementID;
+
+    battleStatus->unk_82 = 5;
+    battleStatus->unk_434 = &D_802943C0;
+    if (battleStatus->unk_83 == 0) {
+        battleStatus->actionSuccess = 0;
+        return ApiStatus_DONE2;
+    }
+    func_80268858();
+
+    actionCommandStatus->actionCommandID = ACTION_COMMAND_MEGA_SHOCK;
+    actionCommandStatus->state = 0;
+    actionCommandStatus->unk_6C = 0x1E;
+    actionCommandStatus->unk_60 = 0;
+    actionCommandStatus->barFillLevel = 0;
+    actionCommandStatus->unk_48 = 0;
+    actionCommandStatus->unk_68 = 0;
+    actionCommandStatus->hudElementX = -0x30;
+    actionCommandStatus->hudElementY = 0x50;
+
+    elementID = create_hud_element(&D_80108B28);
+    actionCommandStatus->hudElements[0] = elementID;
+    set_hud_element_flags(elementID, 0x82);
+    set_hud_element_render_pos(elementID, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(elementID, 0);
+
+    elementID = create_hud_element(&D_8029231C);
+    actionCommandStatus->hudElements[2] = elementID;
+    set_hud_element_flags(elementID, 0x82);
+    set_hud_element_render_pos(elementID, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY);
+    set_hud_element_render_depth(elementID, 0);
+
+    elementID = create_hud_element(&D_80108AFC);
+    actionCommandStatus->hudElements[1] = elementID;
+    set_hud_element_render_pos(elementID, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY + 0x1C);
+    set_hud_element_render_depth(elementID, 0);
+    set_hud_element_flags(elementID, 0x82);
+
+    elementID = create_hud_element(&D_80292974);
+    actionCommandStatus->hudElements[4] = elementID;
+    set_hud_element_render_pos(elementID, actionCommandStatus->hudElementX, actionCommandStatus->hudElementY + 0x1C);
+    set_hud_element_render_depth(elementID, 0);
+    set_hud_element_flags(elementID, 0x82);
+
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802A91A0_42DBB0(Evt* script, s32 isInitialCall) {
+    s16 arg;
```

Comment:
Extremely minor, but I prefer to have the "common" stuff first and have the other declarations below them - could you move this below the first 3?

Reply:
Yup, done.